### PR TITLE
[13.0][MIG] sale_stock_picking_note: Migration to 13.0

### PR DESCRIPTION
**Pull request by joao-p-marques**
@Tecnativa
TT25948

ping @carlosdauden @chienandalu

### CODE REVIEW

**Review by HaraldPanten — APPROVED** [review]
Functional review using runbot 👍

File: sale_stock_picking_note/tests/test_sale_stock_picking_note.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+class TestSaleStockPickingNote(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestSaleStockPickingNote, cls).setUpClass()
+        cls.partner = cls.env["res.partner"].create({"name": "Mr. Odoo",})
```
Comment (carlosdauden) [inline sale_stock_picking_note/tests/test_sale_stock_picking_note.py]:
```suggestion
        cls.partner = cls.env["res.partner"].create({"name": "Mr. Odoo"})
```

File: sale_stock_picking_note/tests/test_sale_stock_picking_note.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        cls.product = cls.env["product.product"].create(
+            {"name": "Test product", "type": "product",}
```
Comment (carlosdauden) [inline sale_stock_picking_note/tests/test_sale_stock_picking_note.py]:
```suggestion
            {"name": "Test product", "type": "product"}
```

File: sale_stock_picking_note/tests/test_sale_stock_picking_note.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        )
+        cls.order = cls.env["sale.order"].create(
+            {
+                "partner_id": cls.partner.id,
+                "order_line": [
+                    (0, 0, {"product_id": cls.product.id, "product_uom_qty": 1,})
```
Comment (carlosdauden) [inline sale_stock_picking_note/tests/test_sale_stock_picking_note.py]:
```suggestion
                    (0, 0, {"product_id": cls.product.id, "product_uom_qty": 1})
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]
Fix travis error:
E231 missing whitespace after ','

**Comment by joao-p-marques** [comment]
Thank you @carlosdauden. Changes made.

**Review by carlosdauden — APPROVED** [review]
Unrelated travis error

**Comment by victoralmau** [comment]
@joao-p-marques Travis errors need solved to review

IMO exist some options about Travis error:
- Incompatibility with `sale_procurement_group_by_line` addon (need to define in manifest.py)
- Improve code about addon `sale_procurement_group_by_line` (https://github.com/OCA/sale-workflow/blob/13.0/sale_procurement_group_by_line/model/sale.py#L30-L120) to allow skip validation with some value in context.
- Improve test with something like this: https://github.com/OCA/sale-workflow/blob/13.0/sale_procurement_group_by_line/tests/test_sale_procurement_group_by_line.py#L84-L87

File: sale_stock_picking_note/tests/test_sale_stock_picking_note.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            {"name": "Test product", "type": "product"}
+        )
+        cls.order = cls.env["sale.order"].create(
+            {
+                "partner_id": cls.partner.id,
+                "order_line": [
+                    (0, 0, {"product_id": cls.product.id, "product_uom_qty": 1})
+                ],
+            }
+        )
```
Comment (victoralmau) [inline sale_stock_picking_note/tests/test_sale_stock_picking_note.py:22]:
```suggestion
        super(TestSaleStockPickingNote, cls).setUpClass()        
        cls.partner = cls.env["res.partner"].create({"name": "Mr. Odoo"})
        cls.product = cls.env["product.product"].create(
            {"name": "Test product", "type": "product"}
        )
        cls.order = cls.env["sale.order"].create(
            {
                "partner_id": cls.partner.id                
            }
        )
        cls.group_id = cls.env["procurement.group"].create(
            {
                "move_type": "one", 
                "sale_id": cls.order.id, 
                "name": cls.order.name
            }
        )
        cls.order_line_1 = cls.env["sale.order.line"].create(
            {
                "order_id": cls.order.id,
                "product_id": cls.product.id,
                "product_uom_qty": 1,
                "name": cls.product.name,
                "procurement_group_id: cls.group_id.id
            }
        )
```

Comment (victoralmau) [reply sale_stock_picking_note/tests/test_sale_stock_picking_note.py:22]: Maybe it solve Travis errors related to `sale_procurement_group_by_line`

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by joao-p-marques** [comment]
Thanks for the review @victoralmau 
However, taking a look at the latest builds from Travis, there seems to be a problem with `sale_procurement_group_by_line`, indeed, that is affecting all builds: https://travis-ci.com/github/OCA/sale-workflow/jobs/407234444 
I tried changing my test as suggested but I didn't improve the situation...

**Comment by joao-p-marques** [comment]
Issue opened about this: https://github.com/OCA/sale-workflow/issues/1304

**Comment by rousseldenis** [comment]
@joao-p-marques Could you rebase when #1305 is merged ?

**Comment by pedrobaeza** [comment]
You can rebase now

**Comment by victoralmau** [comment]
Please, check Travis error

**Comment by joao-p-marques** [comment]
Thank you @victoralmau 
Travis errors are related to prettier, and I think it might be related to https://github.com/pre-commit/pre-commit/issues/1674.
Seems like this was fixed in the 13.0 branch, so I rebased and pushed.

**Review by victoralmau — APPROVED** [review]

**Pull request merged**